packages/smooth_app/lib/pages/product/common/product_list_popup_items.dart

```diff
@@ -216,6 +222,112 @@ class ProductListPopupOpenInWeb extends ProductListPopupItem {
+class ProductListPopupExport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_export;
+
+  @override
+  IconData getIconData() => Icons.download;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.export;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    final String listName = ProductQueryPageHelper.getProductListLabel(
+      productList,
+      AppLocalizations.of(context),
+    );
+    final String fileName =
+        "${listName.replaceAll(' ', '-').toLowerCase()}_${DateTime.now().millisecondsSinceEpoch}.csv";
+    final StringBuffer csv = StringBuffer('Barcode,Name,Brand\n');
+
+    for (final String barcode in productList.barcodes) {
+      final Product? product = await DaoProduct(localDatabase).get(barcode);
+      if (product != null) {
+        final String name = (product.productName ?? '').replaceAll(';', '');
+        final String brand = (product.brands ?? '').replaceAll(';', '');
+        csv.write('"$barcode";"$name";"$brand"\n');
+      }
+    }
+
+    unawaited(
+      Share.shareXFiles(
+        <XFile>[
+          XFile.fromData(
+            utf8.encode(csv.toString()),
+            name: '$fileName.csv',
+            mimeType: 'text/csv',
+          ),
+        ],
+        fileNameOverrides: <String>['$fileName.csv'],
+      ),
+    );
+
+    return null;
+  }
+}
+
+/// Popup menu item for the product list page: import list.
+class ProductListPopupImport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_import;
+
+  @override
+  IconData getIconData() => Icons.upload;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.import;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    FilePicker.platform.pickFiles(
+      type: FileType.custom,
+      allowedExtensions: <String>['csv'],
+    ).then((final FilePickerResult? result) {
+      if (result == null) {
+        return;
+      }
+      final File file = File(result.files.single.path!);
+      final String content = file.readAsStringSync();
+      final List<String> lines = content.split('\n');
+      final List<String> barcodes = <String>[];
+      for (final String line in lines) {
+        final List<String> parts = line.split(';');
+        if (parts.length < 3) {
+          continue;
+        }
+        final String barcode = parts[0].replaceAll('"', '');
+        barcodes.add(barcode);
```

Comment:
Not sure it's that safe to do that. In the "normal" life we add products we already know - they are already related `Product`s in the local database.
Adding unknown barcodes? To be tested.

Reply:
Not sure exactly how the local database works, but if I transfer my data from the old native apps to smooth_app, would the products already be there?

Reply:
> if I transfer my data from the old native apps to smooth_app, would the products already be there?

I guess so, but it's not the point.
I assume that importing/exporting barcodes is a way to say "hello my friend, have a look at those healthy products!". But if you only have barcodes, you don't have the product. And the local list will be populated with locally unknown products.
The import should include product downloads.

Reply:
I think there's something I'm not getting. ProductLists only contains barcodes and no products, but does it require the associated product to be present in the local database? Shouldn't the page displaying a product download it whenever it is not available locally?

Reply:
I'd prefer the solution where you download the products at import time, and if you can't download you don't import into the list.
I have no idea what a list with barcodes without local products would mean. Could be interesting, you can try if you want, but I suspect it will be a pain in the neck in the whole app, for no added value.